packages/scm/src/browser/scm-service.ts

```diff
@@ -0,0 +1,325 @@
+    readonly onDidRemoveRepository: Event<ScmRepository>;
+
+    readonly repositories: ScmRepository[];
+    selectedRepository: ScmRepository | undefined;
+    readonly onDidChangeSelectedRepositories: Event<ScmRepository>;
+
+    registerScmProvider(provider: ScmProvider): ScmRepository;
+}
+
+export interface ScmProvider extends Disposable {
+    readonly label: string;
+    readonly id: string;
+    readonly handle: number;
+    readonly contextValue: string;
+
+    readonly groups: ScmResourceGroup[];
+
+    readonly onDidChangeResources: Event<void>;
+
+    readonly rootUri?: string;
+    readonly count?: number;
+    readonly commitTemplate?: string;
+    readonly onDidChangeCommitTemplate?: Event<string>;
+    readonly onDidChangeStatusBarCommands?: Event<ScmCommand[]>;
+    readonly acceptInputCommand?: ScmCommand;
+    readonly statusBarCommands?: ScmCommand[];
+    readonly onDidChange: Event<void>;
+
+    getOriginalResource(uri: URI): Promise<URI | undefined>;
+}
+
+export interface ScmResourceGroup {
+    readonly handle: number,
+    readonly sourceControlHandle: number,
+    readonly resources: ScmResource[];
+    readonly provider: ScmProvider;
+    readonly label: string;
+    readonly id: string;
+    readonly hideWhenEmpty: boolean | undefined;
+    readonly onDidChange: Event<void>;
+}
+
+export interface ScmResource {
+    readonly handle: number;
+    readonly groupHandle: number;
+    readonly sourceControlHandle: number;
+    readonly group: ScmResourceGroup;
+    readonly sourceUri: URI;
+    readonly decorations?: ScmResourceDecorations;
+    readonly selected?: boolean;
+
+    open(): Promise<void>;
+}
+
+export interface ScmResourceDecorations {
+    icon?: string;
+    tooltip?: string;
+    strikeThrough?: boolean;
+    faded?: boolean;
+
+    source?: string;
+    letter?: string;
+    color?: string;
+}
+
+export interface ScmCommand {
+    id: string;
+    text: string;
+    tooltip?: string;
+    command?: string;
+}
+
+export interface InputValidation {
+    message: string;
+    type: 'info' | 'success' | 'warning' | 'error';
+}
+
+export interface InputValidator {
+    (value: string): Promise<InputValidation | undefined>;
+}
+
+export namespace InputValidator {
+    /**
+     * Type for the validation result with a status and a corresponding message.
+     */
+    export type Result = Readonly<{ message: string, type: 'info' | 'success' | 'warning' | 'error' }>;
+
+    export namespace Result {
+
+        /**
+         * `true` if the `message` and the `status` properties are the same on both `left` and `right`. Or both arguments are `undefined`. Otherwise, `false`.
+         */
+        export function equal(left: Result | undefined, right: Result | undefined): boolean {
+            if (left && right) {
+                return left.message === right.message && left.type === right.type;
+            }
+            return left === right;
+        }
+
+    }
+}
+
+export interface ScmInput {
```

Comment:
Could you review input for unused API and make sure that they are used properly or deleted?

Reply:
done